### ISSUE TITLE
Crash: Tried to pop to a view controller that doesn't exist.

### ISSUE THREAD

**Issue by tiwoc** [helpscout/beacon-ios-sdk#132]
**Technical Details:**

- Beacon install method: 
    [ ] Carthage, 
    [x] CocoaPods, 
    [ ] Manual
- SDK version: 2.1.1
- iOS OS version: 14.1
- Device make/model: iPhone 12 Pro
- Impacted user count: one user out of a small group of beta testers

**Actual outcome**

Beacon crashed the app with the following crash report:

```
Hardware Model:     iPhone13,3
Role:               Foreground
OS Version:         iOS 14.1

NSInternalInconsistencyException: Tried to pop to a view controller that doesn't exist.

0  CoreFoundation    ___exceptionPreprocess
1  libobjc.A.dylib   _objc_exception_throw
2  CoreFoundation    +[NSException raise:format:arguments:]
3  Foundation        -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:]
4  UIKitCore         ___57-[UINavigationController popToViewController:transition:]_block_invoke
5  UIKitCore         -[UINavigationController popToViewController:transition:]
6  Beacon            -[HSBeaconAskSelectionController chatEndedWithNoAvailableAgents]
7  Beacon            Beacon.ChatViewController.noAgentsAvailable() -> ()
8  Beacon            closure #1 () -> () in Beacon.ChatViewController.chatViewModelStateUpdated() -> ()
9  Beacon            reabstraction thunk helper from @escaping @callee_guaranteed () -> () to @escaping @callee_unowned @convention(block) () -> ()
10 libdispatch.dylib __dispatch_call_block_and_release
11 libdispatch.dylib __dispatch_client_callout
12 libdispatch.dylib __dispatch_main_queue_callback_4CF
13 CoreFoundation    ___CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
14 CoreFoundation    ___CFRunLoopRun
15 CoreFoundation    _CFRunLoopRunSpecific
16 GraphicsServices  _GSEventRunModal
17 UIKitCore         -[UIApplication _run]
18 UIKitCore         _UIApplicationMain
19 [redacted]        main (main.m:16:16)
20 libdyld.dylib     _start
```

Looks like `-[HSBeaconAskSelectionController chatEndedWithNoAvailableAgents]` called `-[UINavigationController popToViewController:transition:]` with a view controller that was not on the navigation controller's stack.

**Expected outcome**

Beacon should... not crash the app 🙃

**Steps to reproduce**

Unknown

**Comment by jotaemepereira** [comment]
👋 Hi @tiwoc! We were able to reproduce your crash and we created this [internal ticket](https://helpscout.atlassian.net/browse/IOSSDK-433) for tracking purposes. We will let you know as soon as it gets fixed. Thanks!

**Comment by ScottPetit** [comment]
Hey 👋  @tiwoc,

Thanks for the write up and easy repro steps.  Just wanted to let you know we fixed this in the latest version (2.1.2).  You can read more about what's in this release in our [Changelog](https://github.com/helpscout/beacon-ios-sdk/blob/master/CHANGELOG.md).
